File: leetcode/java/TwoSum.java
import java.util.HashMap;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> comps = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {

            int comp = target - nums[i];
            if (comps.containsKey(comp)) {
                return new int[] { comps.get(comp), i };
            }

            comps.put(nums[i], i);
        }

        return null;
    }
}

File: leetcode/js/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    const comps = new Map();

    for (let i = 0; i < nums.length; i++) {
        
        let comp = target - nums[i];
        if (comps.has(comp)) {
            return [comps.get(comp), i];
        }

        comps.set(nums[i], i);
    }
};

